packages/core/strapi/lib/services/entity-service/index.js

```diff
@@ -178,19 +181,22 @@ const createDefaultImplementation = ({ strapi, db, eventHub, entityValidator })
 
     // TODO: wrap in transaction
     const componentData = await updateComponents(uid, entityToUpdate, validData);
+    const entityData = updatePipeline(
+      Object.assign(omitComponentData(model, validData), componentData),
+      {
+        contentType: model,
+      }
+    );
 
     let entity = await db.query(uid).update({
       ...query,
       where: { id: entityId },
-      data: updatePipeline(Object.assign(omitComponentData(model, validData), componentData), {
-        contentType: model,
-      }),
+      data: entityData,
     });
 
     // TODO: upload the files then set the links in the entity like with compo to avoid making too many queries
-    // FIXME: upload in components
     if (files && Object.keys(files).length > 0) {
-      await this.uploadFiles(uid, entity, files);
+      await this.uploadFiles(uid, Object.assign(entityData, entity), files);
```

Comment:
Nice, this works well for me. I'm just wondering if we could use `entityData` directly in the uploadFiles function? To me it seems to already contain the properties of entity + the component fields we need for this fix. There may be something I'm missing though, WDYT?

Reply:
I don't really know.
I have done this way to change the fewer things I could in the function and keep the current behavior.
Do we want the function to return the entity + component ?